cmake/Platform/Core/BoardPropertiesReader.cmake

```diff
@@ -47,24 +47,28 @@ function(_recursively_replace_properties BOARD_ID PROPERTY_VALUE_VAR)
    # something like ${foo} is not matched as it could be a shell variable 
    # or a cmake variable or whatever, but not a Arduino property.
    #
-   while("${${PROPERTY_VALUE_VAR}}" MATCHES "[^\$]{([^}]*)}")
-      set(variable "${CMAKE_MATCH_1}")
+   while("${${PROPERTY_VALUE_VAR}}" MATCHES "(^|[^\$]){([^}]*)}")
```

Comment:
The regex failed when variables occured directly at the start of the value string.